### PR TITLE
Export mscclpp GpuBuffer to dlpack format

### PR DESCRIPTION
**Pull request by Binyang2014**
For mscclpp, to use nvls we require the buffer is allocated by mscclpp::GpuBuffer. Due to cupy doesn't support bfloat16 yet, we export the raw buffer to dlpack format.
User can use this feature to create buffer with type supported by pytorch
```python
buffer = RawGpuBuffer(1024 * 2) # 2 for bfloat16
dl_pack = buffer.to_dlpack(str(torch.bfloat16))
tensor = torch.utils.dlpack.from_dlpack(dl_pack)
```

### CODE REVIEW

**Review by chhwang — CHANGES_REQUESTED** [review]
Looks good to me, but we would need to support multi-dimensional allocation directly. If users convert 1D to multi-dim, it may incur a local copy which is not intended here.

File: python/mscclpp/gpu_utils_py.cpp
[the remark this answers is not in the document]
Comment (Binyang2014) [reply python/mscclpp/gpu_utils_py.cpp:1]: Covered to draft, If torch memory pool is good enough will close this PR

**Review by chhwang — APPROVED** [review]
LGTM
